File: src/main/java/com/refinedmods/refinedstorage/api/autocrafting/craftingmonitor/ICraftingMonitorListener.java
package com.refinedmods.refinedstorage.api.autocrafting.craftingmonitor;

import com.refinedmods.refinedstorage.api.autocrafting.ICraftingManager;

/**
 * Add this listener to a {@link ICraftingManager} to listen to crafting task changes.
 */
public interface ICraftingMonitorListener {
    /**
     * Called when this listener is attached to a {@link ICraftingManager}.
     */
    void onAttached();

    /**
     * Called when any task changes.
     */
    void onChanged();
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/autocrafting/craftingmonitor/ICraftingMonitorListener.kt
package com.refinedmods.refinedstorage.api.autocrafting.craftingmonitor

import com.refinedmods.refinedstorage.api.autocrafting.ICraftingManager

/**
 * Add this listener to a [ICraftingManager] to listen to crafting task changes.
 */
interface ICraftingMonitorListener {
    /**
     * Called when this listener is attached to a [ICraftingManager].
     */
    fun onAttached()

    /**
     * Called when any task changes.
     */
    fun onChanged()
}